game/plugin/shops/src/shop.plugin.kts

```diff
@@ -0,0 +1,46 @@
+
+import org.apollo.game.message.handler.ItemVerificationHandler
+import org.apollo.game.message.impl.ItemActionMessage
+import org.apollo.game.message.impl.NpcActionMessage
+import org.apollo.game.model.entity.Mob
+import org.apollo.game.scheduling.ScheduledTask
+
+fun Mob.shop(): Shop? = SHOPS[definition.id]
+
+start {
+    ItemVerificationHandler.addInventory(Interfaces.SHOP_CONTAINER) { it.interactingMob?.shop()?.inventory }
+    ItemVerificationHandler.addInventory(Interfaces.INVENTORY_CONTAINER, PlayerInventorySupplier())
+
+    it.schedule(object : ScheduledTask(Shop.RESTOCK_INTERVAL, false) {
+        override fun execute() = SHOPS.values.distinct().forEach(Shop::restock)
+    })
+}
+
+on { NpcActionMessage::class }
+    .then {
+        val npc = it.world.npcRepository.get(index)
+        val shop = npc.shop() ?: return@then
+
+        if (shop.action == option) {
```

Comment:
It'd be good if we intercepted this message and set "option" to a normalized string value.